docs/docs/mail.md

```diff
@@ -1,345 +1,180 @@
+configure the size of your application's messages to a reasonable upper limit.
+
+To avoid the host guessing information from message timing, you can send empty Mail items even when you have nothing 
+to say. For example, if an enclave is running an auction between users, and you wish to hide who won, the enclave 
+can send a "you won" or "you lost" message to every client.
+
+## Comparison to a classical public key infrastructure
+
+Conclave uses Conclave Mail to connect clients to enclaves because of the following reasons:
+
+1. Conclave Mail can work with a small [Trusted Computing Base (TCB)](https://en.wikipedia.org/wiki/Trusted_computing_base) 
+   when compared to HTTPS. A small TCB removes several common security problems at the origin.
+
+2. The certificate-based architecture of SSL/TLS doesn't gel with enclave-based computing. Conclave Mail uses the 
+   better-suited Noise protocol framework. The Noise protocol delivers most of the functionalities of Transport 
+   Layer Security (TLS) in a cleaner, simpler, and modular manner.
+
+   The TLS protocol is reliant on a certificate-based security architecture, which does not suit enclave-based 
+   computing. Certificates don't make sense for enclaves because enclaves are all about *measurements* and *remote 
+   attestations*.
+   
+   If you use TLS, a client communicating with an enclave needs to extract a remote attestation from a 
+   pseudo-certificate. As Conclave Mail uses the Noise protocol, you can provide a byte array instead of a 
+   certificate as part of the handshake.
+
+   In Conclave, the remote attestation data is represented as an
+   [`EnclaveInstanceInfo`](api/-conclave%20-core/com.r3.conclave.common/-enclave-instance-info/index.html) object. The
+   Noise protocol and Conclave doesn't restrict how you get the remote attestation. For example, you can
+   send remote attestation to the client by returning it in another API, publishing it on a web server, and putting it 
+   into a distributed hash table, a network drive, or a message queue.
+
+   Following are the benefits of the Noise protocol over an HTTPS/REST architecture:
+
+   * Eliminate complexity from enclave code.
+   * Avoid hacks like pseudo-certificates.
+   * Enable enclaves to talk to each other even if they can't be loaded simultaneously.
+   * Move session management *and expiry* out of the enclave.
+   * Get other benefits like reusing MQ brokers, integrating with Corda flows, storing messages to databases, and 
+     supporting M-to-1 inbound message collection.
+
+
+3. Conclave Mail's approach to enclave restarts and enclave upgrades is more suited than a classical architecture 
+   that requires a database to ensure session persistence.
+
+4. The primary reason to use HTTPS/REST is the availability of tools and libraries. However, none of these 
+   tools or libraries support SGX remote attestation. You must modify or adjust these tools in complex ways,
+   invalidating most benefits.
```

Comment:
The second sentence can be softened as follows:
"In order to make use of these preexisting tools for remote attestation, they must be used in unusual ways that they were not designed for, which introduces design complexities and reduces their usefulness."

Reply:
Rewrote: "To use these preexisting tools for remote attestation, you must modify them in unusual ways that they were not designed for, which introduces design complexities and reduces their usefulness."